### PR TITLE
F gui tests

### PR DESCRIPTION
**Pull request by mikramarc**
Simple test for checking if the progress bar is being filled

### CODE REVIEW

**Comment by AndriyPt** [comment]
why do you need test launch file here? It can be done without launch file. Maybe this link will help you http://wiki.ros.org/Packages#Python

File: basic/sr_gui_hand_calibration/test/test_hand_calibration.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class TestHandCalibration(unittest.TestCase):
+
+    def setUp(self):
+        self.app = QApplication(sys.argv)
+        self._widget = QWidget()
+        ui_file = os.path.join(rospkg.RosPack().get_path('sr_gui_hand_calibration'), 'uis', 'SrHandCalibration.ui')
+        loadUi(ui_file, self._widget)
+
+        f = open(rospy.get_param('mock_file'), "w+")
+        f.write("""{'sr_calibrations': [\n""" +
```
Comment (AndriyPt) [inline basic/sr_gui_hand_calibration/test/test_hand_calibration.py]: are you modifying package file from test?

**Review by AndriyPt — CHANGES_REQUESTED** [review]

File: basic/sr_gui_hand_calibration/test/test_hand_calibration.py (hunk 1 above)
Comment (mikramarc) [reply basic/sr_gui_hand_calibration/test/test_hand_calibration.py]: No, I am just creating a temporary, dummy (mock) file that is required by the function to be properly called. This file is only for testing and is remove and the tearDown stage

File: basic/sr_gui_hand_calibration/test/test_hand_calibration.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+class TestHandCalibration(unittest.TestCase):
+
+    def setUp(self):
+        self.app = QApplication(sys.argv)
+        self._widget = QWidget()
+        ui_file = os.path.join(rospkg.RosPack().get_path('sr_gui_hand_calibration'), 'uis', 'SrHandCalibration.ui')
+        loadUi(ui_file, self._widget)
+
+        self.mock_file_path = '/tmp/mock_file.yaml'
```
Comment (AndriyPt) [inline basic/sr_gui_hand_calibration/test/test_hand_calibration.py]: please read https://docs.python.org/3/library/tempfile.html#examples

**Review by AndriyPt — CHANGES_REQUESTED** [review]

File: basic/sr_gui_hand_calibration/test/test_hand_calibration.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+class TestHandCalibration(unittest.TestCase):
+
+    def setUp(self):
+        self.app = QApplication(sys.argv)
+        self._widget = QWidget()
+        ui_file = os.path.join(rospkg.RosPack().get_path('sr_gui_hand_calibration'), 'uis', 'SrHandCalibration.ui')
+        loadUi(ui_file, self._widget)
+
+        self.mock_calibration_file = tempfile.NamedTemporaryFile()
```
Comment (AndriyPt) [inline basic/sr_gui_hand_calibration/test/test_hand_calibration.py]: There is constructor argument `delete` set it to `False` and delete file in teardown :)

**Review by AndriyPt — CHANGES_REQUESTED** [review]

File: basic/sr_gui_hand_calibration/test/test_hand_calibration.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        self.app = QApplication(sys.argv)
+        self._widget = QWidget()
+        ui_file = os.path.join(rospkg.RosPack().get_path('sr_gui_hand_calibration'), 'uis', 'SrHandCalibration.ui')
+        loadUi(ui_file, self._widget)
+
+        self.mock_file = tempfile.NamedTemporaryFile(delete=False)
+        self.mock_file.write("""{'sr_calibrations': [\n""" +
+                             """["mock", [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]],\n""" +
+                             """]}""")
+        self.mock_file.seek(0)
```
Comment (AndriyPt) [inline basic/sr_gui_hand_calibration/test/test_hand_calibration.py]: close file here. I guess seek is not needed?

**Review by AndriyPt — CHANGES_REQUESTED** [review]

**Review by AndriyPt — APPROVED** [review]

**Pull request merged**